### PR TITLE
API guide for single line diagram

### PR DESCRIPTION
**Pull request by floriand-e2r**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?**
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Docs update

### CODE REVIEW

File: pages/documentation/developer/repositories/powsybl-single-line-diagram.md [hunk 1]
```diff
@@ -6,11 +6,13 @@ This [repository](https://github.com/powsybl/powsybl-single-line-diagram) provid
 
 ## Features
 
-- creation of single line diagrams for given voltage levels
+- creation of single line diagrams for given voltage levels or substations
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram.md:9]: I think there are more features in that repository:
- a small demo app,
- some extensions
- several layout algorithm

Could you try to pick the main features and complete this list?

File: pages/documentation/developer/repositories/powsybl-single-line-diagram.md [hunk 2]
```diff
@@ -6,11 +6,13 @@ This [repository](https://github.com/powsybl/powsybl-single-line-diagram) provid
...
 
 ## Getting started
 
-- [Guide 1]() - Create a single line diagram on a simple example
+- [Guide 1](powsybl-single-line-diagram_guide1.md) - Create single line diagrams on a simple example
+- [Guide 2](powsybl-single-line-diagram_guide2.md) - Create single line diagrams based on CGMES sample files
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram.md:14]: Move the guide in the specific section. Looking the the description, the guides look like API guides. So you should move the page in `developer/api_guides/single-ligne-diagram` folder, and reference them in the corresponding index (if it exists)

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
+# Generating my first single line diagrams - Guide 1
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:1]: Add the layout:
```
---
layout: default
---

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+In this guide we create a few single line diagrams of a test network with [powsybl-single-line-diagram](powsybl-single-line-diagram.md).
+
+
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:5]: Remove this extra empty line

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+First of all, we need to add a Maven dependency for the Network model and three additional ones for this example: two for the `Network` test case, one for simple logging capabilities:
+
+```xml
+<dependency>
+    <groupId>com.powsybl</groupId>
+    <artifactId>powsybl-iidm-impl</artifactId>
+    <version>3.7.1</version>
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:12]: I don't remember how it's done elsewhere: do we hard-code in the documentation the version?

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+In order to be able to generate the diagrams, we need to add the single line diagram core module dependency:
+```xml
+<dependency>
+    <groupId>com.powsybl</groupId>
+    <artifactId>powsybl-single-line-diagram-core</artifactId>
+    <version>1.7.2</version>
+</dependency>
+```
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:43]: Maybe we should move this in the previous section: add all the required dependencies once.

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+Before generating diagrams we need to initialize a few layout parameters:
+
+```java
+// "Convergence" style component library
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:48]: "Convergence" is a RTE concept. We should rename this style or use a different one?

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+ComponentLibrary componentLibrary = new ResourcesComponentLibrary("/ConvergenceLibrary");
+
+// fully automatic layout
+VoltageLevelLayoutFactory voltageLevelLayoutFactory = new PositionVoltageLevelLayoutFactory(new PositionByClustering());
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:52]: You should explain somewhere what you mean by fully automatic layout. That probably mean that other layout are available.

(Note part of this PR)

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+// create default parameters for the SVG layout
+LayoutParameters layoutParameters = new LayoutParameters();
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:55]: You should explain somewhere what parameters you can set.

(Note part of this PR)

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+```
+
+## Generating a voltage level diagram
+
+We can now generate a SVG for voltage level "N":
+```java
+// create diagram for the voltage level "N"
+VoltageLevelDiagram voltageLevelDiagram = VoltageLevelDiagram.build(new NetworkGraphBuilder(network), "N", voltageLevelLayoutFactory, false);
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:63]: You should extract the false value to a variable: we don't understand what is the fourth parameter.

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+
+// generate SVG
+voltageLevelDiagram.writeSvg("",
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:66]: What is the first parameter (empty string)?

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+    new DefaultSVGWriter(componentLibrary, layoutParameters),
+    new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
+    new NominalVoltageDiagramStyleProvider(network),
+    Paths.get("/tmp/n.svg"));
+```
+
+We obtain the following SVG:
+
+![N_voltageLevel](img/example_n.svg)
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:75]: Don't forget to change the URL

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+voltageLevelDiagram.writeSvg("",
+    new DefaultSVGWriter(componentLibrary, layoutParameters),
+    new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
+    new NominalVoltageDiagramStyleProvider(network),
+    Paths.get("/tmp/c.svg"));
+```
+ 
+leading to the following SVG diagram:
+
+![C_voltageLevel](img/example_c.svg)
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:93]: Same remark

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+// fully automatic layout
+VoltageLevelLayoutFactory voltageLevelLayoutFactory = new PositionVoltageLevelLayoutFactory(new PositionByClustering());
+
+// create default parameters for the SVG layout
+LayoutParameters layoutParameters = new LayoutParameters();
+```
+
+## Generating a voltage level diagram
+
+We can now generate a SVG for voltage level "N":
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:60]: Maybe you should replace `"N"`by `N`

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+    new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
+    new NominalVoltageDiagramStyleProvider(network),
+    Paths.get("/tmp/n.svg"));
+```
+
+We obtain the following SVG:
+
+![N_voltageLevel](img/example_n.svg)
+
+Similarly, we could generate a SVG for voltage level "C":
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:77]: Same remark

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+    new NominalVoltageDiagramStyleProvider(network),
+    Paths.get("/tmp/c.svg"));
+```
+ 
+leading to the following SVG diagram:
+
+![C_voltageLevel](img/example_c.svg)
+
+## Generating a substation diagram
+In order to build the diagram for the whole substation, named "A", containing both voltage levels displayed previously, we need to build the corresponding `SubstationDiagram`:
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:96]: Same remark about `"A"`

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md [hunk 17]
```diff
@@ -0,0 +1,114 @@
...
+```java
+// create diagram for the substation A
+SubstationDiagram substationDiagram = SubstationDiagram.build(
+    new NetworkGraphBuilder(network), "A", new HorizontalSubstationLayoutFactory(),
+    voltageLevelLayoutFactory, false);
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:101]: Same remark about the `false` parameter

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 3 above)
Comment (mathbagu) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md:1]: I think the title is not ok, we should give contextual information `Single Line Diagram - Generating my first diagram`?

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide2.md [hunk 18]
```diff
@@ -0,0 +1,110 @@
+# Generating my first single line diagrams - Guide 2
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide2.md:1]: Same remark

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide2.md [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+    new DefaultSVGWriter(componentLibrary, layoutParameters),
+    Paths.get("/tmp/a.svg"),
+    new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
+    new NominalVoltageDiagramStyleProvider(network)
+);
+```
+
+We then obtain the following SVG file representing the whole "Amsterdam" substation with its four voltage levels:
+
+![Amsterdam_substation](img/example_Amsterdam_substation.svg)
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram_guide2.md:110]: This second guide is exactly the same as the previous one except the case file? Maybe you can merge them to make only one.

**Review by mathbagu — CHANGES_REQUESTED** [review]
You should move the image in a subdirectory. If you look to the other pages, the images are stored in a `img/<name>` folder where `<name>` is the name of the page in which the image is used.

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide2.md (hunk 19 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide2.md]: Indeed! I merged them into one API guide named SVG writing.

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 11 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: The prefix for all IDs of the diagram; I added a variable with a comment to underline it.

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 10 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: I added a variable with a comment to explain that parameter.

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 9 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: I added a reference to the API guide I will write next.
All these default parameters might not be visible soon I hope, once issues [#60](https://github.com/powsybl/powsybl-single-line-diagram/issues/60) and [#46](https://github.com/powsybl/powsybl-single-line-diagram/issues/46) are solved

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 8 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: I added a reference to the API guide about SVG Layouts I will write next.

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 6 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: I merged all the dependencies of two guides into one section

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 7 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: You're right, I was thinking we should name it "Default" instead. I'll create an issue on powsybl-single-line-diagram

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 5 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: No, indeed, it's with a property settings for instance in the loadflow tutorial. Corrected.

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 7 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: We discussed it this morning with the other committers, they don't agree as the library is using the exact same components as RTE's Convergence and therefore keeping that name is decent. Nonetheless Convergence name will not appear in this guide soon:
- there should be soon a new default component library,
- in the next API changes I'm thinking of introducing a new method to generate a SVG without instantiating your own ComponentLibrary (the default ComponentLibrary will be chosen then).
Maybe we should continue this discussion in powsybl-single-line-diagram?

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 3 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: As it is a (single) API guide at the end, I chose a less tutorial-ish title: "Single Line Diagram - SVG Writing"

File: pages/documentation/developer/repositories/powsybl-single-line-diagram.md (hunk 1 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram.md]: I detailed the main features I know of, but I'm not sure I've included them all as I'm still discovering features those days! I've added a TODO, could you have a quick look @geofjamg to check? I've listed them below:

- creation of single line diagrams for given voltage levels, substations or zones in SVG format
- diagrams customization
     - several layout algorithms to generate the diagrams
     - several layout parameters to adjust the rendering
     - use your own component library or modify the existing
- small demo app to display the diagrams (note that this app is mainly for debug purposes and therefore is not included in the release)
- extensions to force positions in the diagram

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 20]
```diff
@@ -0,0 +1,214 @@
+---
+layout: default
+---
+
+# Single Line Diagram - SVG Writing
+
+The `com.powsybl.sld.svg.DefaultSVGWriter` class of [powsybl-single-line-diagram](../repositories/powsybl-single-line-diagram.md) is an implementation of `SVGWriter` interface which allows to generate SVG files representing single line diagrams of a substation, a voltage level or a zone.
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:7]: Typo: an implementation of **the** `SVGWriter` ?

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 21]
```diff
@@ -0,0 +1,214 @@
+---
+layout: default
+---
+
+# Single Line Diagram - SVG Writing
+
+The `com.powsybl.sld.svg.DefaultSVGWriter` class of [powsybl-single-line-diagram](../repositories/powsybl-single-line-diagram.md) is an implementation of `SVGWriter` interface which allows to generate SVG files representing single line diagrams of a substation, a voltage level or a zone. 
+We are showing in this guide how to create some single line diagrams
+- first from a test network,
+- then from a CGMES file.
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:10]: Is it necessary to create a list. Maybe a sentence is better:
> We are showing in this guide how to create some single line diagrams first from a test network, then from a CGMES file.

Add a link from CGMES to the format.

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 22]
```diff
@@ -0,0 +1,214 @@
...
+
+*Note: the 1.8.0-SNAPSHOT version of powsybl-single-line-diagram was used to generate the SVGs of this guide. Because of that, you might experience some differences with the examples shown, which will be corrected by 1.8.0 release (expected: 2020-11-26).*
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:12]: Remove this sentence and add the `do-not-merge` label. I think it's better to wait for the next release instead of talking about a development version.

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 23]
```diff
@@ -0,0 +1,214 @@
...
+    </dependency>
+    <dependency>
+        <groupId>org.slf4j</groupId>
+        <artifactId>slf4j-simple</artifactId>
+        <version>${slf4j.version}</version>
+    </dependency>
+</dependencies>
+
+<properties>
+    <powsybl.sld.version>1.7.2</powsybl.sld.version>
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:60]: This is not consistent with the sentence above: you say that this guide depends on v1.8.0

You should also fix the powsybl.core version to v3.8.0

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 24]
```diff
@@ -0,0 +1,214 @@
...
+    <powsybl.core.version>3.7.1</powsybl.core.version>
+    <slf4j.version>1.7.22</slf4j.version>
+</properties>
+```
+
+Here are some details about these dependencies:
+- `powsybl-single-line-diagram-core` is the core module of single-line-diagram,
+- `powsybl-iidm-impl` is for the Network model,
+- `powsybl-iidm-test` is for loading the test Network,
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:69]: The **N** (in upper case) is not necessary. in the rest of this page you use `Network`. Both lower case or `Network` are OK (you choose!)

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 25]
```diff
@@ -0,0 +1,214 @@
...
+- `powsybl-config-test` is for loading the test configuration (see [configuration API guide](configuration.md)),
+- `powsybl-cgmes-conversion` and `powsybl-triple-store-impl-rdf4j`  are for importing a CGMES file,
+- `slf4j-simple` allows you to have simple logging capabilities.
+
+### Layout parameters
+For both example networks, we need to initialize a few layout parameters before generating diagrams (see [SVG layouts]() API guide for more info):
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:75]: Not sure to understand what you mean by
> For both example networks

`For both examples` or `For both networks` seems sufficient?

I would also change the end to:
> see [SVG layouts guide]() for more info

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 26]
```diff
@@ -0,0 +1,214 @@
...
+
+```java
+// Instantiating the default style component library
+ComponentLibrary componentLibrary = new ResourcesComponentLibrary("/ConvergenceLibrary");
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:79]: Could you also create an issue to rename Convergence... I don't understand why the Convergence word still appear

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 27]
```diff
@@ -0,0 +1,214 @@
...
+
+// fully automatic layout
+VoltageLevelLayoutFactory voltageLevelLayoutFactory = new PositionVoltageLevelLayoutFactory(new PositionByClustering());
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:82]: Could you create an issue in this repository to ask for an explanation of the different positioning algorithm?

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 28]
```diff
@@ -0,0 +1,214 @@
...
+
+// display line ID instead of line name
+boolean usename = false;
+
+// prefix used for the IDs of all diagram components
+String prefix = "";
+```
+
+## Diagrams from a test network
+We first load the node/breaker test `Network` we are interested in:
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:95]: We first **create** ?

Load means from a file, so it's strange here

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 29]
```diff
@@ -0,0 +1,214 @@
...
+```java
+Network network = FictitiousSwitchFactory.create();
+```
+
+### Generating a voltage level diagram
+We generate a `VoltageLevelDiagram` for voltage level N and then the corresponding SVG diagram file:
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:101]: Emphase the N
> votage level `N`

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 30]
```diff
@@ -0,0 +1,214 @@
...
+voltageLevelDiagram.writeSvg(prefix
+    new DefaultSVGWriter(componentLibrary, layoutParameters),
+    new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
+    new NominalVoltageDiagramStyleProvider(network),
+    Paths.get("/tmp/n.svg"));
+```
+
+We obtain the following SVG:
+
+![N_voltageLevel](img/single-line-diagram_svg-basics/example_n.svg)
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:116]: I think you should create a sub-folder:
`img/single-line-diagram/svg-basics/example_n.svg`

Same for the documentation pages. Don't forget to ensure the consistency between the name of the page, and the folder where you put the images

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 31]
```diff
@@ -0,0 +1,214 @@
...
+    new NominalVoltageDiagramStyleProvider(network),
+    Paths.get("/tmp/c.svg"));
+```
+ 
+leading to the following diagram:
+
+![C_voltageLevel](img/single-line-diagram_svg-basics/example_c.svg)
+
+### Generating the substation diagram
+In order to build the diagram for the whole substation, named A, containing both voltage levels displayed previously, we need to build the corresponding `SubstationDiagram`:
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:137]: Emphase the `A`

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 32]
```diff
@@ -0,0 +1,214 @@
...
+    new DefaultSVGWriter(componentLibrary, layoutParameters),
+    Paths.get("/tmp/a.svg"),
+    new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters),
+    new NominalVoltageDiagramStyleProvider(network)
+);
+```
+
+We then obtain the following wider SVG file:
+
+![C_voltageLevel](img/single-line-diagram_svg-basics/example_sub_a.svg)
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:155]: The alternate text is not OK (copy/paste issue)?

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 33]
```diff
@@ -0,0 +1,214 @@
...
+
+## Diagrams from a CGMES file
+
+First of all, we need to download sample files from ENTSO-E [here](https://docstore.entsoe.eu/Documents/CIM_documents/Grid_Model_CIM/TestConfigurations_packageCASv2.0.zip)
```
Comment (mathbagu) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:159]: I think we have factories to create small microgrid (to be checked). I would prefer to be independent from external resources, specially if they decide to remove the file one day

File: pages/documentation/developer/repositories/powsybl-open-loadflow.md [hunk 34]
```diff
@@ -7,10 +7,12 @@ The PowSyBl Open Load Flow [repository](https://github.com/powsybl/powsybl-open-
 ## Features
 
 - a load flow solver
+- a security analysis (beta version)
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-open-loadflow.md:10]: Why these changes appear here?

File: pages/documentation/developer/repositories/powsybl-single-line-diagram.md [hunk 35]
```diff
@@ -6,11 +6,20 @@ This [repository](https://github.com/powsybl/powsybl-single-line-diagram) provid
...
-- creation of single line diagrams for given voltage levels
+- creation of single line diagrams for given voltage levels, substations or zones in SVG format
+- diagrams customization
+     - several layout algorithms to generate the diagrams
+     - several layout parameters to adjust the rendering
+     - use your own component library or modify the existing
+- small demo app to display the diagrams (note that this app is mainly for debug purposes and therefore is not included in the release)
+- extensions to force positions in the diagram 
+
+// TODO: complete the list
```
Comment (mathbagu) [inline pages/documentation/developer/repositories/powsybl-single-line-diagram.md:17]: Do not use this syntax, use the `<span style="color: red">TODO: complete the list</span>` syntax like everywhere.
`//` doesn't comment anything if it's what you're trying to do

**Review by mathbagu — CHANGES_REQUESTED** [review]

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md (hunk 21 above)
Comment (floriand-e2r) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md]: Ok, changed. I linked the website page grid/formats/cim-cgmes.md, but maybe you meant the link to CGMES specifications?

File: pages/documentation/developer/repositories/powsybl-open-loadflow.md (hunk 34 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-open-loadflow.md]: Removed, this had nothing to do here!

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md (hunk 27 above)
Comment (floriand-e2r) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:77]: Created (#176)

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md (hunk 26 above)
Comment (floriand-e2r) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:74]: I'm not sure you saw [my comment above](https://github.com/powsybl/powsybl.github.io/pull/161#discussion_r516601965), as you don't mention it... If you disagree on what's written then I'll create an issue to discuss about it.

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 8 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: Issue #176 created to keep track of that

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 9 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: Issue #177 created to keep track of that

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md (hunk 33 above)
Comment (floriand-e2r) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:154]: I understand, but what is pointed out here is how to create a SVG file from a CGMES file. If we use a factory to create that grid we're losing that part. It's quite easy to import such a file, and it is not a sld feature, sure... but I think we should document it anyway, don't you?
If we want to be independent from external resources, maybe we could host that CGMES file (144ko), or another one?

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md [hunk 36]
```diff
@@ -57,22 +52,22 @@ First of all, we need to add some Maven dependencies in our `pom.xml` file:
...
 
 Here are some details about these dependencies:
 - `powsybl-single-line-diagram-core` is the core module of single-line-diagram,
-- `powsybl-iidm-impl` is for the Network model,
-- `powsybl-iidm-test` is for loading the test Network,
+- `powsybl-iidm-impl` is for the network model,
+- `powsybl-iidm-test` is for loading the test network,
 - `powsybl-config-test` is for loading the test configuration (see [configuration API guide](configuration.md)),
 - `powsybl-cgmes-conversion` and `powsybl-triple-store-impl-rdf4j`  are for importing a CGMES file,
 - `slf4j-simple` allows you to have simple logging capabilities.
```
Comment (annetill) [inline pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:68]: Can you add a link to https://www.powsybl.org/pages/documentation/developer/artifacts.html for powsyblb artifacts ?

File: pages/documentation/developer/repositories/powsybl-open-loadflow.md [hunk 37]
```diff
@@ -7,12 +7,10 @@ The PowSyBl Open Load Flow [repository](https://github.com/powsybl/powsybl-open-
 ## Features
 
 - a load flow solver
-- a security analysis (beta version)
```
Comment (annetill) [inline pages/documentation/developer/repositories/powsybl-open-loadflow.md]: Why do you remove that?

File: pages/documentation/developer/repositories/powsybl-open-loadflow.md [hunk 38]
```diff
@@ -7,12 +7,10 @@ The PowSyBl Open Load Flow [repository](https://github.com/powsybl/powsybl-open-
...
 
 ## Getting started
 
 - [Guide 1]() - Run a load flow computation
-- [Guide 2]() - Run a security analysis
```
Comment (annetill) [inline pages/documentation/developer/repositories/powsybl-open-loadflow.md]: Why do you remove that ?

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md (hunk 33 above)
Comment (annetill) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:154]: Note that this file are in the repo https://github.com/powsybl/powsybl-cgmes-conformity-assessments

File: pages/documentation/developer/repositories/powsybl-open-loadflow.md (hunk 37 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-open-loadflow.md]: In fact I added that line by mistake in this PR, therefore I removed it later (thanks to previous review).

**Review by annetill — CHANGES_REQUESTED** [review]
Some small changes required. It is a bit pity that this kind of really useful documentation is deep in the web site. I think that a link in tutorials section will be nive too. The functional documentation will follow ?

File: pages/documentation/developer/repositories/powsybl-open-loadflow.md (hunk 38 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-open-loadflow.md]: I'm sorry but I don't see any change on open loadflow file in this PR at the end.

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md (hunk 33 above)
Comment (floriand-e2r) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:154]: Link to file hosted on our github added

File: pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md (hunk 7 above)
Comment (floriand-e2r) [reply pages/documentation/developer/repositories/powsybl-single-line-diagram_guide1.md]: [PR](https://github.com/powsybl/powsybl-single-line-diagram/pull/203) created on powsybl-single-line-diagram

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md (hunk 33 above)
Comment (floriand-e2r) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md]: "internal" file linked instead of entsoe external link.

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md (hunk 26 above)
Comment (floriand-e2r) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:74]: [PR](https://github.com/powsybl/powsybl-single-line-diagram/pull/203) created on powsybl-single-line-diagram to have a default constructor

Comment (geofjamg) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:74]: Hiding Convergence word by creating a default constructor is not really a good option it just hard code Convergence in the default constructor...
I do not see any problem with this name, the fact is it is Convergence style components. And I would not have any problem too if someone wants to add a "PowerFactory" library that looks like PowerFactory style....

**Review by annetill — APPROVED** [review]

File: pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md (hunk 26 above)
Comment (mathbagu) [reply pages/documentation/developer/api_guide/single-line-diagram_svg-writing.md:74]: PowerFactory is a well known software, but Convergence is not. As non RTE or Coreso user, I don't understand what Convergence is, and why the default theme is called Convergence. This is my point

**Review by mathbagu — APPROVED** [review]
